spacy_llm/ty.py

```diff
@@ -120,6 +112,16 @@ def parse_responses(
         """
 
 
+_Task_contra = TypeVar("_Task_contra", bound=LLMTaskProtocol, contravariant=True)
+
+
+class TaskResponseParserProtocol(Protocol[_Task_contra]):
+    def __call__(
+        self, task: _Task_contra, docs: Iterable[Doc], responses: Iterable[Any]
+    ) -> Iterable[Any]:
+        ...
```

Comment:
What's the benefit of this over
```suggestion

class TaskResponseParserProtocol:
    def __call__(
        self, task: LLMTaskProtocol, docs: Iterable[Doc], responses: Iterable[Any]
    ) -> Iterable[Any]:
        ...
```
?

Reply:
(Also `docs: Iterable[Doc]` shouldn't be in there, but that's a detail)

Reply:
Ah, because we have a generic protocol in this case. Got it.